server/server/views/registration.py

```diff
@@ -0,0 +1,57 @@
+"""Viws for registration user and confirm registration user in system"""
+from passlib.hash import pbkdf2_sha256
+from passlib.totp import generate_secret
+from pyramid.view import view_config
+from pyramid_mailer.mailer import Mailer
+from pyramid_mailer.message import Message
+from pyramid.response import Response
+from pyramid.httpexceptions import HTTPNotFound
+
+from server.models.user import User
+from server.models.user_status import UserStatus
+from server.models.role import Role
+
+
+@view_config(route_name='registration', renderer='json')
+def registration_view(request):
+    """Registration view
+
+    This function get email and password from json request, check if email
+    isn't in db - add email and password into db, set status_id to 'Non_active',
+    generate and send url-token to user email."""
+    json = request.json_body
+    user_query = User.get_one(request, email=request.json_body['email'])
+    if user_query is None:
+        User.add_user(request, email=request.json['email'],
+                      password=pbkdf2_sha256.hash(request.json['password']),
+                      url_token=generate_secret(),
+                      status_id=UserStatus.get_status_id(request, status="Non_active").id)
+        token = User.get_one(request, email=request.json['email'])
+        mailer = request.mailer
+        message = Message(subject="confirm email",
+                          sender="asstelite@gmail.com",
+                          recipients=[json['email']],
+                          body=request.current_route_url(token.url_token))
```

Comment:
Did you test it? According to documentation it generates URL for current route, which in your case is `registration`. I believe you need to use `route_url` instead ([doc](https://docs.pylonsproject.org/projects/pyramid/en/1.9-branch/narr/urldispatch.html#generating-route-urls))